knowledge_graph/network_class.py

```diff
@@ -11,47 +14,31 @@ class Network:
 
        Completes a depth-first search for the ontology and return edges in
        the component reachable from source.
+
+    Sample Usage
+    ------------
+        onto = get_ontology(onto_path).load()
+        node_network = Network(onto, source)
+        node_network.dfs_labeled_edges()
+        df = pd.DataFrame(node_network.edge_triplets,
+                     columns=["subject", "object", "predcate"])
     """
 
     def __init__(self, ontology, source=None):
         self.ontology = ontology
-        self.result = []
+        self.edge_triplets = []
         self.visited = set()
         self.node_family = []
         self.class_family = []
         if source:
             self.source = source
         else:
             self.source = None
+        # Add labels the ontology object in a way that makes them pythonicly accessible through . invocation method.
         obj_props = list(self.ontology.object_properties())
-        self.obj_properties = self.make_alias_names_for_properties(obj_props)
         annot_props = list(self.ontology.annotation_properties())
-        self.annot_properties = self.make_alias_names_for_properties(annot_props)
-
-    def give_alias(self, property_object):
-        """Adds labels the ontology object in a way that makes them pythonicly accessible through . invocation method.
-
-        Parameters
-        ----------
-        property_object: ontology property object to make pythonicly accessible
-        """
-        label_name = property_object.label[0]
-        label_name = label_name.replace("/", "_or_")
-        label_name = label_name.replace(" ", "_")
-        label_name = label_name.replace(":", "_")
-        property_object.python_name = label_name
-        return label_name
-
-    def make_alias_names_for_properties(self, properties):
-        """Adds labels the ontology object in a way that makes them pythonicly accessible through . invocation method.
-
-        Parameters
-        ----------
-        properties: list of ontology property objects to make pythonicly accessible
-        accessible_names = list of properties now accessible via . invocation method
-        """
-        new_names = [self.give_alias(x) for x in properties]
-        return new_names
+        self.obj_properties = [give_alias(x) for x in obj_props]
```

Comment:
I see the benefit of only having the give_alias function in the ontology_processing_utils... but then that would always have to be loaded in in order for the Network class to work right because the network class doesn't have the give_alias built in, but rather calls upon other code? Just trying to make it easier for others who want to use the Network class to not have to go searching for dependency code to include in their scripts to use the Network class.

Reply:
Either way, it should only live in one place. My concern was that we're exposing ourselves to bugs if `give_alias` is changed in one file where it's defined but not the other, but it's expected to behave the same way in both places. Prior to this refactor, `give_alias` was defined on this class but also in `make_network.py`, from which it was imported into `make_graph`. Seemed to me that the version that was attached to the class was supposed to be the same function, so I made it that way. Considering how it was being distributed across three files, I felt it made more sense to move it to utils to be safe. If it makes more sense to keep it attached to the Network class, we could make it a static method. 

The way I think about classes, it's less about not digging through code than it is logical collections of things. If that aliasing is something that is only meaningful in the context of this process, maybe it makes sense to keep it attached to the graph like this. However, if downstream we might want to take a string and convert it to an alias of the kind we expect to find in the graph, then I'm not sure it makes sense to attach it to Network.

Reply:
I trust your judgement here. Just please check your code still works with the Docker file pipeline. I think some other parts of the app code use give_alias, maybe the visualize ontology code or the main app code? I think it's smart to move give_alias to ontology_processing_utils just please check no other errors occur when trying to build the docker image and run the API that the front end uses.

Reply:
Been running the processing pipeline from within docker for my tests. No errors reported by flask, but I also wasn't actually checking the site. I'll poke around the site and visualization to be sure. I think I was also pretty diligent making sure that any functions I removed weren't being invoked elsewhere in the codebase (by searching the repo in github).